tests/unit/test_helpers.h

```diff
@@ -75,6 +76,39 @@ inline bool isOversubscribed() {
   return num_ranks > CMAKE_DETECTED_MAX_NUM_NODES;
 }
 
+/**
+ * Get a unique filename based on the unit test name.
+ */
+inline std::string getUniqueFilename(
+  const std::string &ext = ""
+) {
+  std::stringstream ss;
+  ss << testing::UnitTest::GetInstance()->current_test_info()->test_suite_name()
+     << "_" << testing::UnitTest::GetInstance()->current_test_info()->name()
+     << ext;
+  std::string str(ss.str());
+  std::size_t pos = 0;
+  while ((pos = str.find("/", pos)) != std::string::npos) {
+    str.replace(pos, 1, "_");
+    ++pos;
+  }
+  return str;
+}
+
+/**
+ * Construct a filename containing the number of ranks so that
+ * concurrently-running tests will not cause file system race conditions.
+ * Do not call this from .nompi.cc tests or from addAdditionalArgs().
+ */
+inline std::string getUniqueFilenameWithRanks(
+  const std::string &ext = ""
```

Comment:
```suggestion
  const std::string& ext = ""
```
(this could probably fit in one line as well)

Reply:
Fixed.